### ISSUE TITLE
ImportError: DLL load failed: The specified module could not be found.  Failed to load the native TensorFlow runtime.

### ISSUE THREAD

**Issue by andmek** [tensorflow/tensorflow#26870]
System information:
Have I written custom code: No
OS Platform and Distribution: Windows 10 Home
Mobile device: None
TensorFlow installed from: binary with pip
TensorFlow version: 1.13.1
Python Version: 3.6.7
Bazel version: not installed
CUDA/cuDNN version: CUDA 10.1, cuDNN 7.5.0.56
GPU model and memory: GeForce GTX 1050 TCC/WDDM
NVIDIA Driver version: 419.35
TensorRT version: 5.0.4.3

Exact command to reproduce:
pip install --force-reinstall tensorflow-gpu
python
import tensorflow as tf

Problem:
I got this error when importing tensorflow. I have checked the environment variables. I have installed Visual Studio Community 2017 and NVIDIA CUDA Visual Studio Integration 10.1.
I have read and tried to follow the solutions from other similar issue (#22794 and #22872), but have not succeeded in fixing the problem.

Log:
C:\Users\Me>python
Python 3.6.7 (v3.6.7:6ec5cf24b7, Oct 20 2018, 13:35:33) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
Traceback (most recent call last):
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "C:\Program Files\Python36\lib\imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "C:\Program Files\Python36\lib\imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 74, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Program Files\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "C:\Program Files\Python36\lib\imp.py", line 243, in load_module
    return load_dynamic(name, filename, file)
  File "C:\Program Files\Python36\lib\imp.py", line 343, in load_dynamic
    return _load(spec)
ImportError: DLL load failed: The specified module could not be found.

Failed to load the native TensorFlow runtime.

**Comment by anto1399** [comment]
@andmek I also run into the same problem. From the official installation page of tenserflow, the CUDA version supported is 10.0, for now 10.1 will give you issues until that is fixed.
So you have to downgrade from 10.1 and follow the setup of the CUDA/cuDNN by updating setting the system path.

**Comment by anto1399** [comment]
@andmek The same issue is on this thread: https://github.com/tensorflow/tensorflow/issues/26364

**Comment by ymodak** [comment]
Closing since its a duplicate of #26364 . Feel free to reopen if the solution provided on that thread doesn't work for you. Thanks!

**Comment by vahidrn98** [comment]
if you have the following configs:
`cuda `and `cudnn `10.0
`tensorflow-gpu`: 2.1
`os` : windows 10
just downgrade to `tensorflow` 2.0.0
this worked for me!
p.s : i think `tensorflow` has high range of sensitivity when it comes to version of the package which is pretty weird and annoying.

**Comment by mihaimaruseac** [comment]
If you get this working by downgrading to 2.0.0 then you can use 2.1.0 by downloading the new MSVC redistributable from Microsoft

**Comment by DrKhalid75** [comment]
there is no solution
tensorflow and its errors causing that I loss my PhD
fuck

**Comment by mihaimaruseac** [comment]
@DrKhalid75 there is a solution, simple search through the duplicates of this (since everyone encountering this just creates a duplicate) would have shown you that there are 4 likely causes of this error:

1. You need to install the MSVC 2019 redistributable. This is exactly the comment above your comment
2. Your CPU does not support AVX2 instructions. Here, you can build from source.
3. Your CPU/Python is on 32 bits. Here, unfortunately there is no solution.
4. There is a library that is in a different location/not installed on your system that cannot be loaded. Here, you have to find out which library is not found/does not exist using Windows SysInternal tools or similar and then create a new separate issue.

In any case, even if TF does not work on your machine, you can use it on Colab. There is no need to claim that a phd gets lost because of a software issue.

**Comment by DrKhalid75** [comment]
1. I downloaded MSVC 2019 redistributable and no way
2. My CPU is Intel (R) Core (TM) i5 CPU   M450 @ 2.4 GHz
3. My CPU / Python 64 bit , Python 2.1
4. I formatted my PC and reinstalled it and no way
.
The problem here I'm in Arabian university, the rule is you choose your point of research and work with yourself, solve any problems with yourself, we haven't any support for you, we want publications only, If you meet funding problems or hardware problems it is not our work, and it is forbidden to change your point of research so you must lost your PhD
